src/adapter/graphql/src/mutations/dataset_updates_schedule_mut.rs

```diff
@@ -0,0 +1,251 @@
+
+use kamu_core::GetSummaryOpts;
+use kamu_dataset_update_flow::*;
+use opendatafabric as odf;
+
+use crate::prelude::*;
+use crate::queries::TimeUnit;
+use crate::{utils, LoggedInGuard};
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone)]
+pub struct DatasetUpdatesScheduleMut {
+    dataset_handle: odf::DatasetHandle,
+}
+
+#[Object]
+impl DatasetUpdatesScheduleMut {
+    #[graphql(skip)]
+    pub fn new(dataset_handle: odf::DatasetHandle) -> Self {
+        Self { dataset_handle }
+    }
+
+    #[graphql(guard = "LoggedInGuard::new()")]
+    async fn set_polling_time_delta(
+        &self,
+        ctx: &Context<'_>,
+        paused: bool,
+        every: TimeDeltaInput,
+    ) -> Result<SetUpdateScheduleResult> {
+        self.ensure_expected_dataset_kind(ctx, opendatafabric::DatasetKind::Root)
+            .await?;
+        self.ensure_scheduling_permission(ctx).await?;
+
+        let update_schedule_service = from_catalog::<dyn UpdateScheduleService>(ctx).unwrap();
+
+        let update_state = update_schedule_service
+            .set_schedule(
+                self.dataset_handle.id.clone(),
+                Schedule::TimeDelta(ScheduleTimeDelta {
+                    every: every.into(),
+                }),
+            )
+            .await
+            .map_err(|e| match e {
+                SetScheduleError::Internal(e) => GqlError::Internal(e),
+            })?;
+
+        self.apply_pause_resume_schedule(update_schedule_service.as_ref(), paused, &update_state)
+            .await?;
+
+        Ok(SetUpdateScheduleResult::Success(
+            SetUpdateScheduleResultSuccess { success: true },
+        ))
+    }
+
+    #[graphql(guard = "LoggedInGuard::new()")]
+    async fn set_polling_cron_expression(
+        &self,
+        ctx: &Context<'_>,
+        paused: bool,
+        cron_expression: String,
+    ) -> Result<SetUpdateScheduleResult> {
+        self.ensure_expected_dataset_kind(ctx, opendatafabric::DatasetKind::Root)
+            .await?;
+        self.ensure_scheduling_permission(ctx).await?;
+
+        let update_schedule_service = from_catalog::<dyn UpdateScheduleService>(ctx).unwrap();
+
+        let update_state = update_schedule_service
+            .set_schedule(
+                self.dataset_handle.id.clone(),
+                Schedule::CronExpression(ScheduleCronExpression {
+                    expression: cron_expression,
+                }),
+            )
+            .await
+            .map_err(|e| match e {
+                SetScheduleError::Internal(e) => GqlError::Internal(e),
+            })?;
+
+        self.apply_pause_resume_schedule(update_schedule_service.as_ref(), paused, &update_state)
+            .await?;
+
+        Ok(SetUpdateScheduleResult::Success(
+            SetUpdateScheduleResultSuccess { success: true },
+        ))
+    }
+
+    #[graphql(guard = "LoggedInGuard::new()")]
+    async fn set_throttling(
```

Comment:
Name `set_batching` seems more appropriate to me - as we define how to combine high-frequency updates into batches.

I would actually rename `throttling_period` here to `batching_period` as I think we should use the term `throttling` purely for additional artificial processing delay we will be adding on public platform to save costs.

Reply:
We are using term "reactive" schedule, as a contrast to "proactive", so my name of preference now is `set_reactive`